src/core/manage.py

```diff
@@ -309,38 +310,48 @@ class DictionaryManagement(Command):
     option_list = (
         Option('--upload-cve', dest='opt_cve', action='store_true'),
         Option('--upload-cpe', dest='opt_cpe', action='store_true'),
+        Option('--upload-cve-url', dest='cve_url', required=False),
+        Option('--upload-cpe-url', dest='cpe_url', required=False),
     )
 
-    def run(self, opt_cve, opt_cpe):
+    def run(self, opt_cve, opt_cpe, cve_url, cpe_url):
         from model import attribute
 
-        if (opt_cve):
+        app.logger.debug("opt_cve: {} opt_cpe: {} cve_url: {} cpe_url: {}".format(opt_cve, opt_cpe, cve_url, cpe_url))
+        if opt_cve or (cve_url is not None):
             cve_update_file = getenv('CVE_UPDATE_FILE')
             if cve_update_file is None:
                 app.logger.critical("CVE_UPDATE_FILE is undefined")
                 abort()
 
-            self.upload_to(cve_update_file)
+            if cve_url is not None:
+                self.download_update_file(cve_update_file, cve_url)
+            else:
+                self.upload_to(cve_update_file)
             try:
                 attribute.Attribute.load_dictionaries('cve')
+                app.logger.info("CVE Dictionary updated")
             except Exception:
-                app.logger.critical("File structure was not recognized!")
+                app.logger.exception("File structure was not recognized!")
```

Comment:
you should also catch the Exception and print it to the user so it is easier to debug what failed (`except Exception as ex:` - please use `ex` as the variable so it is consistent with other files). I suggest the use of traceback https://github.com/SK-CERT/Taranis-NG/blob/cca219beb864ed9bcdd6f6e637a230e3f864be04/src/collectors/collectors/web_collector.py#L518

Reply:
This is actually excatly what logger.exception is here for it will print the stacktrace automatically and is usally preferred over calling traceback. See: https://docs.python.org/3/library/logging.html#logging.exception

Reply:
ahh, okay, I did not know about this one. it looks neater than the other code I suggested.

Reply:
+1 for using `logging.exception`

Reply:
if the output is similar and gives enough information to solve an issue, I agree with the use of `logging.exception` instead. we will refactor the rest of the code to stay consistent

Reply:
It outputs the hole traceback, but with level ERROR, not DEBUG as you suggested. If you prefer to have the traceback in DEBUG, I suggest to write your modify `app.logger`'s behaviour centrally to keep the exception handling code simple and easy.

Reply:
I guess errors and Exceptions should be visible whether or not DEBUG is enabled. It makes sense. Debug should be only used for additional, more verbose output.